pkg/network/netlink/consumer.go

```diff
@@ -0,0 +1,404 @@
+package netlink
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"syscall"
+
+	"github.com/pkg/errors"
+
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"golang.org/x/sys/unix"
+
+	"github.com/mdlayher/netlink"
+	"github.com/mdlayher/netlink/nlenc"
+)
+
+const (
+	netlinkCtNew      = uint32(1)
+	ipctnlMsgCtGet    = 1
+	outputBuffer      = 100
+	overshootFactor   = 0.95
+	netlinkBufferSize = 1024 * 1024 // 1Mb
+)
+
+var msgBufferSize int
+
+func init() {
+	msgBufferSize = os.Getpagesize()
+}
+
+var errShortErrorMessage = errors.New("not enough data for netlink error code")
+var errMaxSamplingAttempts = errors.New("netlink socket creation: too many attempts")
+
+// Consumer is responsible for encapsulating all the logic of hooking into Conntrack
+// and streaming new connection events.
+type Consumer struct {
+	conn      *netlink.Conn
+	socket    *Socket
+	pool      *sync.Pool
+	workQueue chan func()
+
+	// throttling
+	samplingRate float64
+	rateLimit    int
```

Comment:
Can we document the units and also the difference between `rateLimit` and `samplingRate`? It's not really clear what the difference is.

Reply:
This is done.